core/dbt/include/global_project/macros/adapters/show.sql

```diff
@@ -0,0 +1,21 @@
+{% macro show(limit) -%}
```

Comment:
What do you think of giving this macro a longer name, like `get_show_sql`? (There is always some slight risk of adding a macro to the global namespace that collides with a macro which users have already defined in their projects.)

For readability, I'd suggest either:
- an inline comment that `compiled_code` is coming from [the context](https://github.com/dbt-labs/dbt-core/blob/d597b80486ebb409685fafb23d972f5361ddf7fc/core/dbt/context/providers.py#L1397)
- adding `compiled_code` and `sql_header` as macro arguments, and passing them in explicitly from `adapter.execute_macro`
```python
            kwargs={
                "compiled_code": model_context["compiled_code"],
                "sql_header": model_context["config"].get("sql_header"),
                "limit": limit
            },
        )
```

Reply:
Great suggestions, will incorporate them + go for passing in macro arguments explicitly :)